src/hybrid_crypto/header.rs

```diff
@@ -20,16 +20,23 @@ use crate::{
 #[derive(Debug, PartialEq, Clone)]
 pub struct Metadata {
     pub uid: Vec<u8>,
-    pub additional_data: Vec<u8>,
+    pub additional_data: Option<Vec<u8>>,
 }
 
 impl Metadata {
     /// Encode the metadata as a byte array
     pub fn as_bytes(&self) -> anyhow::Result<Vec<u8>> {
         let mut bytes = u32_len(&self.uid)?.to_vec();
         bytes.extend(&self.uid);
-        bytes.extend(u32_len(&self.additional_data)?.to_vec());
-        bytes.extend(&self.additional_data);
+        match &self.additional_data {
+            Some(ad_data) => {
+                bytes.extend(u32_len(ad_data)?.to_vec());
+                bytes.extend(ad_data);
+            }
+            None => {
+                bytes.extend([0_u8; 4]);
```

Comment:
proposal, use a const value named something like `EMPTY_ADDITIONAL_DATA` to refer to this 4-byte array.

we need this because parsing needs to have a size right (zero in this case)?